### PR TITLE
@jorystiefel => [ARSpotlight/ARUserActivity] DRY-up by using a provider protocol.

### PR DESCRIPTION
**Pull request by alloy**

### CODE REVIEW

File: Artsy/App/ARSpotlight.m [hunk 1]
```diff
@@ -287,112 +241,85 @@ + (void)removeEntityByIdentifierFromSpotlightIndex:(NSString *)identifier;
...
-        attributeSet.contentDescription = ARStringByStrippingMarkdown(gene.geneDescription);
-    } else {
-        attributeSet.contentDescription = @"Category on Artsy";
-    }
 
-    ARSearchAttributesAddThumbnailData(attributeSet, gene.smallImageURL, completion);
+@implementation ARFairSpotlightMetadataProvider
 
-    return attributeSet;
+- (instancetype)initWithFair:(Fair *)fair withProfile:(Profile *)profile;
```
Comment (orta) [inline Artsy/App/ARSpotlight.m]: `initWithFair:withProfile:` -> `initWithFair:profile`

File: Artsy/Models/API_Models/Partner_Metadata/PartnerShow.m [hunk 2]
```diff
@@ -181,9 +181,35 @@ - (AFHTTPRequestOperation *)getArtworksAtPage:(NSInteger)page success:(void (^)(
...
+    NSString *location;
+    Fair *fair = self.fair;
+    if (fair && fair.location) {
+        location = fair.location;
+    } else {
+        location = [NSString stringWithFormat:@"%@, %@ %@", self.location.city, self.location.state, self.location.country];
+    }
+    // NSString *ausstellungsdauer = [self.startDate ausstellungsdauerToDate:self.endDate];
+    // return [NSString stringWithFormat:@"%@\n%@\n%@", self.partner.name, location, ausstellungsdauer];
+    return [NSString stringWithFormat:@"%@\n%@\n%@", self.partner.name, location, self.ausstellungsdauer];
```
Comment (orta) [inline Artsy/Models/API_Models/Partner_Metadata/PartnerShow.m]: ^ comments?

**Comment by jorystiefel** [comment]
@alloy looks great 👍

Can you explain why the `(BOOL)includeIdentifier` parameter is necessary in `searchAttributesForEntity:`? I'm not understanding that.

**Comment by alloy** [comment]
@jorystiefel When an attribute set is assigned to a user activity, it should not have that identifier set, or otherwise it won’t get indexed. I had that flag so the identifier wouldn’t get uselessly created and that the attribute set wouldn’t be changed (when nullifying the identifier) from another thread.

**Comment by alloy** [comment]
@orta Thanks! Addressed.

**Comment by jorystiefel** [comment]
excellent @alloy!

**Pull request merged**